### DISCUSSION TITLE
Trying to get the project running

### DISCUSSION THREAD

**Discussion by zeke** [nytimes/library#260] [Q&A/Help]
Hello! 👋 

I'm an engineer on the documentation team at GitHub. My colleague @sarahs and I have been watching this project for a while, and now I'm finally trying it out.

I've got this set up:

- [x] App is running on Node.js 10, macOS
- [x] Created a Google API project with Drive and Cloud Store access
- [x] Created a service account 
- [x] Created `server/.auth.json` with credentials file created by Google
- [x] Created OAuth 2 client id
- [x] Created `.env` with all values
- [x] Created a new folder in Drive using my personal Google account and updated related values in `.env`
- [x] Created a new Google Doc in that folder with some filler content

Unfortunately, I'm still seeing a lot of errors in my console. I added them to a gist here: https://gist.github.com/zeke/c2572650c5211985b33aa42106f1ec02

Some highlights from that output:

- 'Cannot find module \'/Users/z/git/forks/library/custom/cache/store\'
- `warn: GOOGLE_APPLICATION_CREDENTIALS was undefined, using default ./.auth.json credentials file...`
- `Cannot find module \'/Users/z/git/forks/library/custom/userAuth\''`
- `Cannot find module \'/Users/z/git/forks/library/custom/csp\`

### Context (Environment)

```
$ node --version
v10.16.2

$ uname -a
Darwin calvisitor-10-105-181-69.calvisitor.1918.berkeley.edu 18.7.0 Darwin Kernel Version 18.7.0: Sat Oct 12 00:02:19 PDT 2019; root:xnu-4903.278.12~1/RELEASE_X86_64 x86_64
```

### Expected Behavior

When running `npm run watch`, I expect to see a local running server.

### Actual Behavior

I see a bunch of errors: https://gist.github.com/zeke/c2572650c5211985b33aa42106f1ec02

### To Reproduce

Follow the checklist above. :)

### Possible Solution

It looks like there are some undocumented expectations in the code about certain local files that need to exist.

Maybe the Google App doesn't have access to the Google Drive folder I created? 

I'd be happy to jump on a video call with someone to pair on this. Once I've got a successfully running installation of `library` I'd be happy to follow up with updates to the documentation to make the setup process more clear for newcomers.

Thanks for reading!

**Comment by isaacwhite** [comment]
Hey @zeke! Thanks for providing such a detailed description of the steps you tried. Many of the traces you mentioned are noncritical, and probably unrelated to the problem you are encountering. (They're often expected, and we should make them less prominent.)

One of the later traces displayed in your console is most likely the cause of the issue:
```
[0] debug: searching for files > 0
[0] warn: failed updating tree 
[0] { message: 'File not found: .',
...
 data:
[0]       { error:
[0]          { errors:
[0]             [ { domain: 'global',
[0]                 reason: 'notFound',
[0]                 message: 'File not found: .',
[0]                 locationType: 'parameter',
[0]                 location: 'fileId' } ],
[0]            code: 404,
[0]            message: 'File not found: .' } },
``` 

That trace comes from the part of the code where we make a request to the Drive API to list the documents, and could result from a lack of permissions. 

Did you share the folder you created with the email of your service account? That's the final piece that grants the service account access.

**Comment by zeke** [comment]
Thanks for the quick response!

> Did you share the folder you created with the email of your service account?

☝️ I had not! Now I have:

![Screen Shot 2019-10-31 at 5 01 54 PM](https://user-images.githubusercontent.com/2289/67993952-37ed2a00-fc00-11e9-9dd6-81c3ac04419d.png)

---

I'm still seeing the same (or similar) errors in the console after restarting the app. Does that invitation need to be accepted somehow by the service account?

**Comment by isaacwhite** [comment]
Normally adding the service account is all that should be needed, though in some cases with shared folders (vs a Shared/Team drive) we've noticed permissions can take a little while to update on Google's side. 

Are you still seeing that `File not found` error in your application logs since you added the account a few hours ago? In particular, the continued presence of this log line would suggest something else could also be wrong:

> warn: failed updating tree

**Comment by zeke** [comment]
Just tried again this morning. Still seeing `warn: failed updating tree`.

**Comment by maxine** [comment]
Hi @zeke ! Sorry for the delayed response. Your setup process looks correct, but based on errors I've seen in the past, this seems like it might be a configuration issue. A few things to check/try:

- Put the Google doc with filler content into a folder nested within your shared folder. Do you still see the `failed updating tree` error message?
- In your `.env` file, is `DRIVE_TYPE` set to `folder` or `shared`? There's conflicting documentation on this (working on updating now!), but it should be set to `folder`. 
- Check that `DRIVE_ID` in `.env` includes the entire hash in the Google Drive folder's url, that `.auth.json` is in the `server/` directory, and that `GCP_PROJECT_ID` in `.env` is the same as `"project_id"` in `.auth.json`

I'll reach out and we can find some time to pair if nothing mentioned above resolves the problem.

**Comment by zeke** [comment]
Hey @maxine thank you for this info. I haven't yet made time to try out your suggestions above, but I intend to, and will report my findings here. Perhaps over the holiday break. :) 🎄

**Comment by blumysden** [comment]
Hi @maxine!  FYI, I'm getting some errors on startup similar to @zeke.  Your suggestion to put the filler content in a nested folder took care of the tree updating errors, thank you very much.  But I do still see the `MODULE_NOT_FOUND` errors:

```
"Cannot find module '/Users/scottblumenthal/workspace/blumysden/library/custom/cache/store'",
...
"Cannot find module '/Users/scottblumenthal/workspace/blumysden/library/custom/userAuth'",
...
"Cannot find module '/Users/scottblumenthal/workspace/blumysden/library/custom/csp'"
```

Pretty sure all the auth stuff you listed above checks out.  And the app is running and displaying my files, so that's good!

(The rest of the setup has gone smoothly.  I'm excited to finally get to play around with my own Library install!)

**Comment by atf-leroy** [comment]
Hey everyone, 

I also been having some problems deploying. I'm getting this error and do not know what to do "There was an issue building your app. This can mean your app.json's project is not a valid Heroku application. Please ensure your app is deployable to Heroku and try again."

I made multiple changes to fix this error but with no result. 
<img width="829" alt="OSXDaily 2024-04-19 om 14 03 53" src="https://github.com/nytimes/library/assets/150681511/0dd88d8f-640a-4fdc-be70-894aab5fa290">
